tests/multi_gcov_aggregator.py

```diff
@@ -0,0 +1,145 @@
+                                    "MODULE",  "PRIVATE", "PUBLIC", "CONTAINS", "INTERFACE",
+                                    "use", "implicit", "namelist",
+                                    "USE", "IMPLICIT", "NAMELIST",
+                                    "else", "end", "case", "continue",
+                                    "ELSE", "END", "CASE", "CONTINUE",
+                                    "!", "#", "&",
+                                    "1","2","3","4","5","6","7","8","9") #format strings
+                        if len(stripped_line) == 0 or stripped_line.startswith(keywords):
+                            count = "-"  # Line is never executable
+                        else:
+                            count = 0    # Line is executable but not covered, possibly due to preprocessor directives
+                    elif count.endswith("*"):  # Branch execution
+                        count = int(count[:-1])
+                    else:
+                        count = int(count)
+
+                    line_number = int(line_number)
+                    # Aggregate counts and keep line content
+                    current_count, current_line_content, directories = self.coverage_data[source_file][line_number]
+
+                    if isinstance(count, int) and count > 0 and directory is not None:
+                        directories.append(directory)
+
+                    if isinstance(count, int):
+                        if isinstance(current_count, int):
+                           current_count += count
+                        else:
+                           current_count = count
+
+                    if line_content != current_line_content and current_line_content != "":
+                        print(f"Warning: Line content mismatch in {source_file} at line {line_number}")
+                        print(f"  Previous line content: {current_line_content}")
+                        print(f"  Current line content: {line_content}")
+                        print(f"  Using previous line content and count for aggregation.")
+                    else:
+                        self.coverage_data[source_file][line_number] = (current_count, line_content, directories)
+
+    def parse_directories(self, directories):
+        """
+        Parse all GCOV files from multiple directories.
+        """
+        for directory in directories:
+            for file_path in glob(os.path.join(directory, "*.gcov")):
+                self.parse_gcov_file(file_path, os.path.basename(directory))
+
+    def save_aggregated_coverage(self, output_directory):
+        """
+        Save the aggregated coverage data for each source file.
+        """
+        full_code_tot = 0
+        full_code_coverage = 0
+        os.makedirs(output_directory, exist_ok=True)
+
+        # sort files alphabetically
+        sorted_files = sorted(list(self.coverage_data.keys()))
```

Comment:
```suggestion
        sorted_files = sorted(self.coverage_data.keys())
```
or even
```suggestion
        sorted_files = sorted(self.coverage_data)
```
but the latter is a bit more ambiguous (although equivalent)